[dillongoh] iP

dillongoh opened the pull request:
### TunaBot
> "I think Duke isn't fishy enough" - Dillon Goh

TunaBot 🐟 is the extra fishy assistant that loves assisting you! It's

- text-based
- says **BLUB!**
- ~~easy~~ _SUPER EASY_ to use

Reasons to use TunaBot:
1. Helps you keeps track of tasks
2. Says **BLUB!**

TunaBot is like this task list but better!
- [x] Task 1
- [ ] Task 2
- [ ] Task 3

Upgrade TunaBot following instructions [here](https://nus-cs2103-ay2223s2.github.io/website/schedule/week3/project.html)!

Heres a peek of the `main` method:
```
public static void main(String[] args) {
        Path savePath = Paths.get("data", "save.txt");
        new TunaBot(savePath).run();
}
```

Commit:
Add Gradle support

dillongoh committed:
modified duke.java to greet, echo and exit

dillongoh committed:
added Add and List features

dillongoh committed:
renamed bot to TunaBot and added Task object to manage tasks

dillongoh committed:
added support for tracking ToDo, Event and Deadline tasks.

dillongoh committed:
added support for testing using redirection technique

dillongoh committed:
modified runtest.sh to new bot name

dillongoh committed:
added support for error handling

dillongoh committed:
updated Task.java, Deadline.java, Event.java

dillongoh committed:
updated EXPECTED.TXT and input.txt to test error handling

dillongoh committed:
added support for deleting tasks

dillongoh committed:
modified input.txt to test delete

dillongoh committed:
Add Save feature to maintain task list after TunaBot is closed

dillongoh committed:
Add support for LocalDateTime format in Deadline and Events

dillongoh committed:
Merge branch 'branch-Level-7'

* branch-Level-7:
  Add Save feature to maintain task list after TunaBot is closed

dillongoh committed:
Merge branch 'branch-Level-8'

* branch-Level-8:
  Add support for LocalDateTime format in Deadline and Events

# Conflicts:
#	src/main/java/TunaBot.java

dillongoh committed:
Fix bugs involving date and savng format

dillongoh committed:
Add Storage object to handle loading and saving tasks to file.

dillongoh committed:
Add Parser class to handle input of commands

dillongoh committed:
Add UI class to handle interactions with user.

dillongoh committed:
Add TaskList class to manage tasks, modify Parser, Storage and TunaBot to use TaskList

dillongoh committed:
Organize classes into packages

dillongoh committed:
Merge remote-tracking branch 'origin/add-gradle-support'

dillongoh committed:
Add Checkstyle support

dillongoh committed:
Add JUnit Tests for Task and Deadline classes

dillongoh committed:
Add JavaDoc comments to the code

dillongoh committed:
Fix code style to comply with coding standard

dillongoh committed:
Add Find command to search for tasks

dillongoh committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/tunabot/Storage.java
#	src/main/java/tunabot/TunaBot.java
#	src/main/java/tunabot/exceptions/InputException.java
#	src/main/java/tunabot/task/Deadline.java
#	src/main/java/tunabot/task/Event.java

dillongoh committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/tunabot/Parser.java

dillongoh committed:
Add basic support for GUI using JavaFX

dillongoh committed:
Format string outputs

dillongoh committed:
Merge branch 'branch-Level-10'

jmestxr commented on src/main/java/tunabot/TunaBot.java:
```diff
@@ -0,0 +1,178 @@
+     * Initializes a TunaBot with the given save path.
+     */
+    public TunaBot() {
+
```
(Small issue) empty line could be removed

jmestxr commented on src/main/java/tunabot/TunaBot.java:
```diff
@@ -0,0 +1,178 @@
+    /**
+     * You should have your own function to generate a response to user input.
+     * Replace this stub with your completed method.
+     */
```
I think this javadocs is from the tutorial. Just remember to replace this with your own :)

jmestxr commented on src/main/java/tunabot/Ui.java:
```diff
@@ -0,0 +1,86 @@
+        return add;
+    }
+
+
```
Perhaps you could standardise the number of newlines between two adjacent methods. E.g. Keep to one newline between add(...) and saveFileProblem(), just like what you did between delete(...) and add(...).

jmestxr commented on src/main/java/tunabot/TunaBot.java:
```diff
@@ -0,0 +1,178 @@
+public class TunaBot extends Application {
+    private static final Scanner s = new Scanner(System.in);
+    private static TaskList tasks;
+    private static boolean toExit = false;
```
toExit should be renamed to isExit to adhere to coding standards.

jmestxr commented on src/main/java/tunabot/task/Deadline.java:
```diff
@@ -0,0 +1,53 @@
+    public String getDeadline() {
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd/MM/yy-HHmm");
+        return deadline.format(formatter);
+    }
```
You could add a newline here to between the two methods for better readability.

jmestxr commented on src/main/java/tunabot/task/Deadline.java:
```diff
@@ -0,0 +1,53 @@
+    @Override
+    public String saveFormat() {
+        return "T;" + this.name + ";" + this.getDone() + ";" + this.getDeadline();
+    }
```
Same here.

jmestxr commented on src/main/java/tunabot/task/Task.java:
```diff
@@ -0,0 +1,62 @@
+     */
+    public void unmarkDone() {
+        this.isDone = false;
+    }
```
Same here. You could add a newline for better readability.

jmestxr commented on src/main/java/tunabot/task/Task.java:
```diff
@@ -0,0 +1,62 @@
+    }
+    public String saveFormat() {
+        return "T;" + this.name + ";" + this.isDone;
+    }
```
Same here. You could add a newline for better readability.

jmestxr reviewed (commented):
Overall, I like how you have adhered to the Java coding standards and quality guidelines most of the time. You could also consider creating Command classes (e.g. AddCommand, DeleteCommand etc) that encapsulates the action of executing the relevant user command, to make your code more OOP.

dillongoh committed:
Add Assertions

There are no use of assert to define assumptions about program state.

Assertions ensure that assumptions about program state are met when certain code is run

Let's add assertions to Parser.java and TunaBot.java to ensure the TaskList object was initialised properly

dillongoh committed:
Improve Code Quality

Code quality have not been reviewed in a long time.

Improving code quality improves readability and understandability of the code

Let's improve code quality by extracting some methods from long classes making use of SLAP.

dillongoh committed:
Merge pull request #1 from dillongoh/branch-A-Assertions

Add Assertions

dillongoh committed:
Merge branch 'master' of https://github.com/dillongoh/ip into branch-A-CodeQuality

dillongoh committed:
Merge pull request #2 from dillongoh/branch-A-CodeQuality

Improve Code Quality

dillongoh committed:
Set up CI

dillongoh committed:
Fix Checkstyle errors

dillongoh committed:
Add an update feature to support editing existing tasks

dillongoh committed:
Fixed checkstyle errors and removed unused imports

dillongoh committed:
Merge pull request #3 from dillongoh/branch-C-Update

Branch c update

pkwangjoo commented on src/main/java/tunabot/Storage.java:
```diff
@@ -0,0 +1,92 @@
+     * @throws IOException if an I/O Exception occurs when creating a new file.
+     */
+    public TaskList load() throws IOException {
+        File saveFile = getSaveFile();
```
Perhaps `saveFile` variable could be named as a noun instead?

pkwangjoo replied:
Perhaps `saveFile` variable could be named as a noun instead?

pkwangjoo commented on src/main/java/tunabot/Ui.java:
```diff
@@ -0,0 +1,86 @@
+        return greeting;
+    }
+
+    public void line() {
```
Should this method name be names in a verb format? perhaps `showLine`

pkwangjoo commented on src/main/java/tunabot/Parser.java:
```diff
@@ -0,0 +1,190 @@
+            return newDeadline(tasks, command);
+        case "find":
+            return find(tasks, command);
+        case"update":
```
A minor nit but should there be a space between the `case` and `"update"`?

pkwangjoo commented on src/main/java/tunabot/Parser.java:
```diff
@@ -0,0 +1,190 @@
+        String[] details = command[1].split(" /name | /from | /by | /to ");
+        try {
+            int index = Integer.parseInt(details[0]) - 1;
+            Task updateTask = tasks.get(index);
```
Perhaps this could be named in a noun format?

dillongoh committed:
Improve the GUI by adding a custom made profile picture for TunaBot and User and using FXML

dillongoh committed:
Fix some formatting issues and GUI not showing entire task list

dillongoh committed:
Add Ui.png

dillongoh committed:
Add User Guide

dillongoh committed:
Update JavaDocs

dillongoh committed:
Update JavaDocs and applied SLAP to some long methods

dillongoh committed:
Remove unnecessary JavaDoc

dillongoh committed:
Remove trailing whitespace

dillongoh committed:
Change README.md to be the User Guide
